### PR TITLE
Back-end changes to implement #1461

### PR DESCRIPTION
**Pull request by Algogator**
## Changes

- Changed User API so response returns current_team and list of teams a user belongs to.
- Changed User model to include relationship to user's current_team and team property now returns current_team
- Added some tests

*Please describe.*  
*If this affects the front-end, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [x] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

File: posthog/api/user.py [hunk 1]
```diff
@@ -23,11 +23,22 @@ def user(request):
...
+    teams = list(request.user.team_set.all().values("name"))
 
     if request.method == "PATCH":
         data = json.loads(request.body)
 
         if "team" in data:
+            if "current_team" in data["team"]:
+                current_team = data["team"].get("current_team", team)
+                if isinstance(current_team, str):
+                    request.user.current_team = request.user.team_set.get(name=current_team)
```
Comment (timgl) [inline posthog/api/user.py:35]: Names aren't unique, we should be using team ids here :)

**Review by timgl — COMMENTED** [review]
This is very cool!

File: frontend/src/layout/Teams.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                    current_team: id,
+                },
+            })
+            toast.success('Team changed')
+        } catch (response) {
+            toast.error(response.error)
+        }
+    }
+
+    return (
```
[the remark this answers is not in the document]
Comment (Algogator) [reply frontend/src/layout/Teams.js:31]: @timgl this is still a little buggy and I have to spend a little more time on it!

**Comment by Twixes** [comment]
Closed in favor of #1562, as to work on a branch of ours instead of @Algogator's. :)

**Pull request closed without merging**